{"path":"lucene/backwards/src/test/org/apache/lucene/index/TestCompoundFile#demo_FSIndexInputBug(Directory,String).mjava","commits":[{"id":"9454a6510e2db155fb01faa5c049b06ece95fab9","date":1453508333,"type":1,"author":"Dawid Weiss","isMerge":false,"pathNew":"lucene/backwards/src/test/org/apache/lucene/index/TestCompoundFile#demo_FSIndexInputBug(Directory,String).mjava","pathOld":"backwards/src/test/org/apache/lucene/index/TestCompoundFile#demo_FSIndexInputBug(Directory,String).mjava","sourceNew":"    private void demo_FSIndexInputBug(Directory fsdir, String file)\n    throws IOException\n    {\n        // Setup the test file - we need more than 1024 bytes\n        IndexOutput os = fsdir.createOutput(file);\n        for(int i=0; i<2000; i++) {\n            os.writeByte((byte) i);\n        }\n        os.close();\n\n        IndexInput in = fsdir.openInput(file);\n\n        // This read primes the buffer in IndexInput\n        byte b = in.readByte();\n\n        // Close the file\n        in.close();\n\n        // ERROR: this call should fail, but succeeds because the buffer\n        // is still filled\n        b = in.readByte();\n\n        // ERROR: this call should fail, but succeeds for some reason as well\n        in.seek(1099);\n\n        try {\n            // OK: this call correctly fails. We are now past the 1024 internal\n            // buffer, so an actual IO is attempted, which fails\n            b = in.readByte();\n            fail(\"expected readByte() to throw exception\");\n        } catch (IOException e) {\n          // expected exception\n        }\n    }\n\n","sourceOld":"    private void demo_FSIndexInputBug(Directory fsdir, String file)\n    throws IOException\n    {\n        // Setup the test file - we need more than 1024 bytes\n        IndexOutput os = fsdir.createOutput(file);\n        for(int i=0; i<2000; i++) {\n            os.writeByte((byte) i);\n        }\n        os.close();\n\n        IndexInput in = fsdir.openInput(file);\n\n        // This read primes the buffer in IndexInput\n        byte b = in.readByte();\n\n        // Close the file\n        in.close();\n\n        // ERROR: this call should fail, but succeeds because the buffer\n        // is still filled\n        b = in.readByte();\n\n        // ERROR: this call should fail, but succeeds for some reason as well\n        in.seek(1099);\n\n        try {\n            // OK: this call correctly fails. We are now past the 1024 internal\n            // buffer, so an actual IO is attempted, which fails\n            b = in.readByte();\n            fail(\"expected readByte() to throw exception\");\n        } catch (IOException e) {\n          // expected exception\n        }\n    }\n\n","bugFix":null,"bugIntro":[],"isBuggy":false,"nexts":[],"revCommit":null},{"id":"90eca6fcb6635ca73ea4fdbe2f57d2033b66d3b6","date":1272983566,"type":4,"author":"Uwe Schindler","isMerge":false,"pathNew":"/dev/null","pathOld":"lucene/backwards/src/test/org/apache/lucene/index/TestCompoundFile#demo_FSIndexInputBug(Directory,String).mjava","sourceNew":null,"sourceOld":"    private void demo_FSIndexInputBug(Directory fsdir, String file)\n    throws IOException\n    {\n        // Setup the test file - we need more than 1024 bytes\n        IndexOutput os = fsdir.createOutput(file);\n        for(int i=0; i<2000; i++) {\n            os.writeByte((byte) i);\n        }\n        os.close();\n\n        IndexInput in = fsdir.openInput(file);\n\n        // This read primes the buffer in IndexInput\n        byte b = in.readByte();\n\n        // Close the file\n        in.close();\n\n        // ERROR: this call should fail, but succeeds because the buffer\n        // is still filled\n        b = in.readByte();\n\n        // ERROR: this call should fail, but succeeds for some reason as well\n        in.seek(1099);\n\n        try {\n            // OK: this call correctly fails. We are now past the 1024 internal\n            // buffer, so an actual IO is attempted, which fails\n            b = in.readByte();\n            fail(\"expected readByte() to throw exception\");\n        } catch (IOException e) {\n          // expected exception\n        }\n    }\n\n","bugFix":null,"bugIntro":[],"isBuggy":false,"nexts":[],"revCommit":null}],"commit2Parents":{"90eca6fcb6635ca73ea4fdbe2f57d2033b66d3b6":["9454a6510e2db155fb01faa5c049b06ece95fab9"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":[],"cd5edd1f2b162a5cfa08efd17851a07373a96817":["90eca6fcb6635ca73ea4fdbe2f57d2033b66d3b6"],"9454a6510e2db155fb01faa5c049b06ece95fab9":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"]},"commit2Childs":{"90eca6fcb6635ca73ea4fdbe2f57d2033b66d3b6":["cd5edd1f2b162a5cfa08efd17851a07373a96817"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":["9454a6510e2db155fb01faa5c049b06ece95fab9"],"9454a6510e2db155fb01faa5c049b06ece95fab9":["90eca6fcb6635ca73ea4fdbe2f57d2033b66d3b6"],"cd5edd1f2b162a5cfa08efd17851a07373a96817":[]},"heads":["cd5edd1f2b162a5cfa08efd17851a07373a96817"],"roots":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"],"pathCommit":null}